multimodal/health_multimodal/image/model/model.py

```diff
@@ -0,0 +1,167 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+from dataclasses import dataclass
+from typing import Any, List, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from pl_bolts.models.self_supervised.resnets import resnet18, resnet50
+from torch import Tensor as T
+
+from .modules import ClassifierHeadType, MLP, MTModel
+
+
+@dataclass
+class ImageModelOutput():
+    img_embedding: torch.Tensor
+    patch_embedding: torch.Tensor
+    projected_global_embedding: torch.Tensor
+    class_logits: torch.Tensor
+    projected_patch_embeddings: torch.Tensor
+
+
+class ImageModel(nn.Module):
+    """
+    Image encoder module
+    """
+
+    def __init__(self,
```

Comment:
How are we planning to let the user know what parameters to use to match our pretrained weights? One lightweight option could be to define a specific variation of the image encoder that is compatible with our pretrained weights.

Here's an example from torchvision:
https://github.com/pytorch/vision/blob/9b5a3fecc72434dbd65148723efe54b28b9728c9/torchvision/models/resnet.py#L11
and
https://github.com/pytorch/vision/blob/9b5a3fecc72434dbd65148723efe54b28b9728c9/torchvision/models/resnet.py#L301

Reply:
I think that'd work. The weights can easily be in blob storage:

```python
model_urls = {
    "resnet18": "https://download.pytorch.org/models/resnet18-f37072fd.pth",
    "resnet34": "https://download.pytorch.org/models/resnet34-b627a593.pth",
    "resnet50": "https://download.pytorch.org/models/resnet50-0676ba61.pth",
    "resnet101": "https://download.pytorch.org/models/resnet101-63fe2227.pth",
    "resnet152": "https://download.pytorch.org/models/resnet152-394f9c45.pth",
    "resnext50_32x4d": "https://download.pytorch.org/models/resnext50_32x4d-7cdf4587.pth",
    "resnext101_32x8d": "https://download.pytorch.org/models/resnext101_32x8d-8ba56ff5.pth",
    "wide_resnet50_2": "https://download.pytorch.org/models/wide_resnet50_2-95faca4d.pth",
    "wide_resnet101_2": "https://download.pytorch.org/models/wide_resnet101_2-32ee1156.pth",
}
```

```python
def _resnet(
    arch: str,
    block: Type[Union[BasicBlock, Bottleneck]],
    layers: List[int],
    pretrained: bool,
    progress: bool,
    **kwargs: Any,
) -> ResNet:
    model = ResNet(block, layers, **kwargs)
    if pretrained:
        state_dict = load_state_dict_from_url(model_urls[arch], progress=progress)
        model.load_state_dict(state_dict)
    return model
```

Note that the TorchVision repo uses [PyTorch Hub](https://pytorch.org/hub/): https://github.com/pytorch/vision/blob/main/hubconf.py

Reply:
This is maybe out of the scope of this PR, we can discuss it in the future.